### PR TITLE
WikiBooks Dataset Processing

### PR DESCRIPTION
**Pull request by PhungVanDuy**

### CODE REVIEW

**Comment by ncoop57** [comment]
Hey @PhungVanDuy looking good! One thing that will need to be changed though is to use our interfaces for code datasets. You can see an example of how this works with the stackexchange dataset: https://github.com/CarperAI/Code-Pile/blob/main/codepile/stackexchange/stackexchange.py

Essentially, we want to follow the same style for all our datasets so that the code is easier to manage and extend

**Comment by PhungVanDuy** [comment]
> Hey @PhungVanDuy looking good! One thing that will need to be changed though is to use our interfaces for code datasets. You can see an example of how this works with the stackexchange dataset: https://github.com/CarperAI/Code-Pile/blob/main/codepile/stackexchange/stackexchange.py
> 
> Essentially, we want to follow the same style for all our datasets so that the code is easier to manage and extend

@ncoop57  I just have a quick refactor, please check.

**Comment by reshinthadithyan** [comment]
Hello, Daica. Thanks for the PR. PRs should go to the `working` branch not the `main`

**Comment by PhungVanDuy** [comment]
> Hello, Daica. Thanks for the PR. PRs should go to the `working` branch not the `main`

Thank for your comment, just changed!

File: codepile/enwikibooks/README.md [hunk 1]
```diff
@@ -0,0 +1,19 @@
+### Download WikiBook dumps dataset 
+
+```bash
+wget https://dumps.wikimedia.org/enwikibooks/20220901/enwikibooks-20220901-pages-meta-current.xml.bz2 -O data/enwikibooks-20220820-pages-meta-current.xml.bz2
```
Comment (ncoop57) [inline codepile/enwikibooks/README.md]: make sure to add `mkdir data` before this incase the person hasn't created the folder

File: codepile/enwikibooks/wikibooks.py [hunk 2]
```diff
@@ -0,0 +1,92 @@
+import os
+import json
+import pandas as pd
+from codepile.dataset import RawDataset, Scraper, Dataset
+
+import requests
+from bs4 import BeautifulSoup
```
Comment (ncoop57) [inline codepile/enwikibooks/wikibooks.py:9]: Make sure to add this dependency to the project.toml file

**Comment by ncoop57** [comment]
@PhungVanDuy looking good! added some comments

**Comment by ncoop57** [comment]
@PhungVanDuy added some comments. I think it is almost ready to merge once the unit test issues are resolved!

**Comment by PhungVanDuy** [comment]
> @PhungVanDuy added some comments. I think it is almost ready to merge once the unit test issues are resolved!

I wrote the unit test [here](https://github.com/PhungVanDuy/Code-Pile/blob/books_dataset/codepile/enwikibooks/unittests.py) can you check it? @ncoop57

**Comment by PhungVanDuy** [comment]
> > @PhungVanDuy added some comments. I think it is almost ready to merge once the unit test issues are resolved!
> 
> I wrote the unit test [here](https://github.com/PhungVanDuy/Code-Pile/blob/books_dataset/codepile/enwikibooks/unittests.py) can you check it? @ncoop57

I just saw your example for the unit test in the main branch. Do you mean I should modify the test code to follow the class style?

**Comment by ncoop57** [comment]
@PhungVanDuy it can follow the class style I created or this function style as shown in this pytest site: https://docs.pytest.org/en/7.1.x/getting-started.html

**Comment by PhungVanDuy** [comment]
> @PhungVanDuy it can follow the class style I created or this function style as shown in this pytest site: https://docs.pytest.org/en/7.1.x/getting-started.html

I just changed the style for tests.py file, let me know if have any problems.

**Comment by ncoop57** [comment]
@PhungVanDuy could you grant me access to make changes to this PR?

You can enable it by following this blog: https://github.blog/2016-09-07-improving-collaboration-with-forks/

**Comment by ncoop57** [comment]
Okay, so fixed up the tests and pyproject.toml. Sadly I can't actually run the tests since they require S3 keys haha. @reshinthadithyan can you run the tests? I am not on the HPC. If not we can just go ahead and merge. It lgtm

**Comment by PhungVanDuy** [comment]
> Okay, so fixed up the tests and pyproject.toml. Sadly I can't actually run the tests since they require S3 keys haha. @reshinthadithyan can you run the tests? I am not on the HPC. If not we can just go ahead and merge. It lgtm

Can we merge after today?, I need some edit readme for data processing.

**Comment by ncoop57** [comment]
@PhungVanDuy I think it is ready to merge. That okay?

**Comment by PhungVanDuy** [comment]
@ncoop57  yes, please

**Pull request merged**